waspc/src/Wasp/Psl/Ast/Schema.hs

```diff
@@ -0,0 +1,39 @@
+module Wasp.Psl.Ast.Schema
+  ( Schema (..),
+    Block (..),
+    getModels,
+    getEnums,
+    getDatasources,
+    getGenerators,
+  )
+where
+
+import Wasp.Psl.Ast.ConfigBlock (ConfigBlock)
+import qualified Wasp.Psl.Ast.ConfigBlock as Psl.ConfigBlock
+import Wasp.Psl.Ast.Enum (Enum)
+import Wasp.Psl.Ast.Model (Model)
+import Prelude hiding (Enum)
+
+data Schema = Schema [Block]
+  deriving (Show, Eq)
+
+data Block
+  = ModelBlock Model
+  | EnumBlock Enum
+  | ConfigBlock ConfigBlock
+  deriving (Show, Eq)
+
+getModels :: Schema -> [Model]
+getModels (Schema blocks) = [model | ModelBlock model <- blocks]
+
+getEnums :: Schema -> [Enum]
+getEnums (Schema blocks) = [enum | EnumBlock enum <- blocks]
+
+getDatasources :: Schema -> [ConfigBlock]
+getDatasources schema = [datasource | datasource@((Psl.ConfigBlock.ConfigBlock Psl.ConfigBlock.Datasource _ _)) <- getConfigBlocks schema]
+
+getGenerators :: Schema -> [ConfigBlock]
+getGenerators schema = [generator | generator@((Psl.ConfigBlock.ConfigBlock Psl.ConfigBlock.Generator _ _)) <- getConfigBlocks schema]
```

Comment:
Ok so you simplified the model here! What is not so great is that you can't be more precise in type now, you now lost that precision, you have to say you are returning config blocks although you know you are really returning datasources. That is why at the end of that big comment I said that maybe it is best to leave it as you had it because it was quite descriptive at the type level.
But ok, probably not a big deal, we can always see about it in the future. This is simpler at least.